app/scripts/views/settings/client_disconnect.js

```diff
@@ -0,0 +1,128 @@
+  const FormView = require('views/form');
+  const ModalSettingsPanelMixin = require('views/mixins/modal-settings-panel-mixin');
+  const SignedOutNotificationMixin = require('views/mixins/signed-out-notification-mixin');
+  const t = require('views/base').t;
+  const Template = require('stache!templates/settings/client_disconnect');
+
+  const REASON_SELECTOR = '.disconnect-reasons';
+  const REASON_HELP = {
+    'lost': t('We\'re sorry to hear about this. You should change your Firefox Account password, and look for ' +
+      'information from your device manufacturer about erasing your data remotely.'),
+    'suspicious': t('We\'re sorry to hear about this. If this was a device you really don\'t trust, you should change your ' +
+      'Firefox Account password, and change any passwords saved in Firefox.')
+  };
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'clients-disconnect',
+    viewName: 'settings.clients.disconnect',
+
+    events: {
+      'change select': 'selectOption',
+      'click': 'closePanelIfDisconnected'
+    },
+
+    initialize () {
+      // user is presented with an option to disconnect device
+      this.toDisconnect = true;
+    },
+
+    beforeRender () {
+      // receive the device collection and the item to delete
+      // if deleted the collection will be automatically updated in the settings panel.
+      let clients = this.model.get('clients');
+      let deviceId = this.model.get('itemId');
+      if (! clients || ! deviceId) {
+        return this.navigate('settings/clients');
+      }
+
+      this.item = clients.get(deviceId);
+    },
+
+    context () {
+      var context = {
+        reasonHelp: this.reasonHelp,
+        toDisconnect: this.toDisconnect
+      };
+
+      if (this.toDisconnect) {
+        context.deviceName = this.item.get('name');
+      }
+
+      return context;
+    },
+
+    afterRender () {
+      // disable the form by default, user must select an option
+      this.disableForm();
+    },
+
+    /**
+     * Called on option select.
+     * If first option is selected then form is disabled.
+     *
+     * @param {Event} event
+     */
+    selectOption (event) {
+      let optionIndex = this.$el.find(event.currentTarget).find(':selected').index();
+      if (optionIndex === 0) {
+        this.disableForm();
+      } else {
+        this.enableForm();
+      }
+    },
+
+    submit () {
+      let item = this.item;
+      let selectedValue = this.$el.find(REASON_SELECTOR).find(':selected').val();
+      this.logViewEvent('submit.' + selectedValue);
+
+      return this.user.destroyAccountClient(this.user.getSignedInAccount(), item)
+        .then(() => {
+          // user has disconnect the device
+          this.toDisconnect = false;
+          this.reasonHelp = REASON_HELP[selectedValue];
+          if (item.get('isCurrentDevice')) {
+            // if disconnected the current device then sign out
+            this._closePanelReturnToClients();
```

Comment:
Calling `_closePanelReturnToClients` then `navigateToSignIn` will call `this.navigate` twice. What happens if you just call `this.navigateToSignIn`, that should close the panel, shouldn't it?